src/mpid/ch4/src/ch4_types.h

```diff
@@ -278,6 +278,17 @@ typedef struct {
 #define MPIDIU_THREAD_MPIDIG_GLOBAL_MUTEX MPIDI_global.m[3]
 #define MAX_CH4_MUTEXES 4
 
+/* VCI */
+typedef struct MPIDI_vci {
+    MPID_Thread_mutex_t lock;   /* lock to protect this VCI */
+    MPIR_Request *lw_req;       /* pre-allocated completed request for this VCI */
+    MPIDI_workq_t workqueue;    /* WorkQ for this VCI */
```

Comment:
Add `int attr` member. Having it make it easy to extend the infrastructure later.

Reply:
Did we decide drop vci attributes for now? I am okay with it.